File: src/mPanel.API/Core/Entities/PanelSettings.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace mPanel.API.Core.Entities;

public class PanelSettings
{
    public int Id { get; private set; } = 1;

    public string Name { get; set; } = "mPanel";
    public string? Url { get; set; }

    public bool AllowRegistration { get; set; } = true;
    public bool AllowAccountSelfDeletion { get; set; } = true;

    public Smtp Smtp { get; set; } = new();
}

[ComplexType]
public class Smtp
{
    public string? Host { get; set; }
    public int Port { get; set; } = 587;
    public string? Username { get; set; }
    public string? Password { get; set; }
    public string? From { get; set; }
}

File: src/mPanel.Web/app/utils/schemas/panel-settings.ts
import * as v from 'valibot'

export const updatePanelSettingsSchema = v.object({
    name: v.pipe(
        v.string(),
        v.trim(),
        v.nonEmpty('Name is required'),
        v.maxLength(32, 'Name is too long'),
    ),
    url: v.optional(v.pipe(v.string(), v.trim(), v.url('Invalid url'))),
    allowRegistration: v.boolean(),
    allowAccountSelfDeletion: v.boolean(),
    smtp: v.object({
        host: v.optional(v.string()),
        port: v.pipe(
            v.number(),
            v.integer(),
            v.minValue(1, 'Port must be greater than 0'),
            v.maxValue(65535, 'Port must be less than 65536'),
        ),
        username: v.optional(v.string()),
        password: v.optional(v.string()),
        from: v.optional(v.pipe(v.string(), v.email('Invalid email'))),
    }),
})

export type UpdatePanelSettingsSchema = v.InferInput<typeof updatePanelSettingsSchema>
